### PR TITLE
Rearrange Modal's layout in accessibility mode

### PR DESCRIPTION
**Pull request by M-i-k-e-l**

### CODE REVIEW

File: src/screensComponents/modal/index.js [hunk 1]
```diff
@@ -39,18 +39,28 @@ export default class Modal extends BaseComponent {
...
-          accessibilityLabel={accessibilityLabel}
-          accessibilityRole="button"
-          style={[styles.touchableOverlay, {backgroundColor: overlayBackgroundColor}]}
-        >
-          <TouchableWithoutFeedback onPress={onBackgroundPress}>
-            <View flex/>
-          </TouchableWithoutFeedback>
-        </View>
-      );
+      if (Constants.accessibility.isScreenReaderEnabled) {
```
Comment (ethanshar) [inline src/screensComponents/modal/index.js]: can we somehow unify those two separate renders? 
for instance you can pass the `useSafeArea` prop to view in order to render SafeAreaView.. 
maybe you can condition the other props.

Comment (M-i-k-e-l) [reply src/screensComponents/modal/index.js]: I've unified them, not sure it's more readable, let me know.

File: src/screensComponents/modal/index.js [hunk 2]
```diff
@@ -39,17 +39,21 @@ export default class Modal extends BaseComponent {
   renderTouchableOverlay() {
     const {overlayBackgroundColor, onBackgroundPress, accessibilityLabel = 'Dismiss'} = this.props;
     if (_.isFunction(onBackgroundPress) || !!overlayBackgroundColor) {
+      const isScreenReaderEnabled = Constants.accessibility.isScreenReaderEnabled;
+      const Container = isScreenReaderEnabled ? SafeAreaView : View;
+      const containerStyle = isScreenReaderEnabled
```
Comment (ethanshar) [inline src/screensComponents/modal/index.js]: You can tidy this a little by doing
`<View style={!isScreenReaderEnabled && [styles.touchableOverlay, {backgroundColor: overlayBackgroundColor}]} ...`

File: src/screensComponents/modal/index.js [hunk 3]
```diff
@@ -39,17 +39,21 @@ export default class Modal extends BaseComponent {
   renderTouchableOverlay() {
     const {overlayBackgroundColor, onBackgroundPress, accessibilityLabel = 'Dismiss'} = this.props;
     if (_.isFunction(onBackgroundPress) || !!overlayBackgroundColor) {
+      const isScreenReaderEnabled = Constants.accessibility.isScreenReaderEnabled;
+      const Container = isScreenReaderEnabled ? SafeAreaView : View;
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/screensComponents/modal/index.js]: Done

File: src/screensComponents/modal/index.js (hunk 2 above)
Comment (M-i-k-e-l) [reply src/screensComponents/modal/index.js]: Done

**Pull request merged**